### PR TITLE
Refactor charon-config.xml template.

### PR DESCRIPTION
**Pull request by hasinidilanka**
**Proposed Changes**

- Remove unused property `pagination-default-count` from the configuration file.

- Hardcoded default values for configurations, that should not be configurable further. 
      
```
patch-supported
documentationUri
bulk-supported
filter-supported
filter-maxResults
changePassword
sort-supported
etag-supported
```

### CODE REVIEW

File: modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/charon-config.xml.j2 [hunk 1]
```diff
@@ -18,13 +18,21 @@
...
 
 <provisioning-config>
-    {% for key,value in identity.charon.properties.items()%}
-    <Property name="{{key}}">{{value}}</Property>
-    {% endfor %}
+    <Property name="user-schema-extension-enabled">{{charon.properties.user_schema_extension.enabled}}</Property>
+    <Property name="bulk-maxOperations">{{charon.properties.bulk_operations.max}}</Property>
+    <Property name="bulk-maxPayloadSize">{{charon.properties.bulk_payload_size.max}}</Property>
+    <Property name="patch-supported">true</Property>
+    <Property name="documentationUri">http://example.com/help/scim.html</Property>
```
Comment (pulasthi7) [inline modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/charon-config.xml.j2]: Let's make this configurable and have wso2 doc link as default

File: modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/charon-config.xml.j2 [hunk 2]
```diff
@@ -18,13 +18,21 @@
...
+    <Property name="changePassword">true</Property>
+    <Property name="sort-supported">false</Property>
+    <Property name="etag-supported">false</Property>
     <authenticationSchemes>
-        {% for scheme in identity.charon.enabled_auth_schemes%}
-        <schema>
-            {% for key,value in identity.charon.authentication_schemes[scheme].items()%}
+        {% for scheme in charon.enabled_auth_schemes%}
+        <schema id="{{loop.index}}">
+            {% for key,value in charon.authentication_schemes[scheme].items()%}
```
Comment (pulasthi7) [inline modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/charon-config.xml.j2]: Seems we won't need to loop the scemas, so let's have the two known schemas only

File: modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/charon-config.xml.j2 (hunk 1 above)
Comment (hasinidilanka) [reply modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/charon-config.xml.j2]: Fixed.

File: modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/charon-config.xml.j2 (hunk 2 above)
Comment (hasinidilanka) [reply modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/charon-config.xml.j2]: Fixed.

File: modules/distribution/src/repository/resources/conf/default.json [hunk 3]
```diff
@@ -162,39 +162,13 @@
...
-      "specUri": "http://www.rfc-editor.org/info/rfc2617",
-      "documentationUri": "http://example.com/help/httpBasic.html",
-      "type": "httpbasic",
-      "primary": false
-    }
-  },
+  "charon.properties.user_schema_extension.enabled": true,
+  "charon.properties.bulk_operations.max": "1000",
+  "charon.properties.bulk_payload_size.max": "1048576",
+  "charon.properties.documentation_uri": "https://docs.wso2.com/display/ISCONNECTORS/SCIM+2.0+Provisioning+Connector",
```
Comment (pulasthi7) [inline modules/distribution/src/repository/resources/conf/default.json]: Have this one instead,
https://docs.wso2.com/display/IS570/apidocs/SCIM2-endpoints/index.html

Comment (hasinidilanka) [reply modules/distribution/src/repository/resources/conf/default.json]: Fixed with [977e133](https://github.com/wso2/product-is/pull/4809/commits/977e13335591ab9112147cd3cd97765c8afd4562)

**Pull request merged**